src/services/RetroGenerationService.ts

```diff
@@ -98,6 +104,52 @@ class RetroGenerationService {
         });
     }
 
+    private adjustStaticTemplateForMoreThan11Tests(template: { workbook: Excel.Workbook, reportTemplate: any}, testResults: any) {
+        const worksheet = template.workbook.getWorksheet(1);
+        const numberOfRowsToBeAdded = testResults.length - 11;
+        for (let i = 39 + numberOfRowsToBeAdded; i >= 28; i--) {
+            const currentRow = worksheet.getRow(i);
+            const rowToBeShifted = worksheet.getRow(i - numberOfRowsToBeAdded);
+            currentRow.height = rowToBeShifted.height;
+            currentRow.border = rowToBeShifted.border;
+            for (let j = 1; j < 17; j++) {
+                const currentCell = currentRow.getCell(j);
+                const cellToBeShifted = rowToBeShifted.getCell(j);
+                currentCell.style = cellToBeShifted.style;
+                currentCell.value = cellToBeShifted.value;
+            }
+            currentRow.commit();
+        }
+    }
+
+    private correctTemplateAfterAdjustment(template: { workbook: Excel.Workbook, reportTemplate: any}, testResults: any) {
+        const documentRequestHeaderCellIndex = 18 + testResults.length;
+        const worksheet = template.workbook.getWorksheet(1);
+        worksheet.getCell(`B${documentRequestHeaderCellIndex}`).value = "Document Requests";
+
+        const siteVisitDetailsWithBordersCellArray = ["B4", "B6", "B7", "E4", "E5", "E6", "E7"];
+        this.addBorders(siteVisitDetailsWithBordersCellArray, worksheet);
+
+        const activityDetailsWithBordersCellArray = this.constructTableCellArray(17, 17 + testResults.length - 1, "B", "M");
+        this.addBorders(activityDetailsWithBordersCellArray, worksheet);
+
+        const documentRequestWithBordersCellArray = this.constructTableCellArray(17 + testResults.length + 2, 17 + testResults.length + 3, "B", "G");
+        this.addBorders(documentRequestWithBordersCellArray, worksheet);
+
+        const healthySafetyIssuesWithBordersCellArray = this.constructTableCellArray(17 + testResults.length + 6, 17 + testResults.length + 7, "B", "G");
+        this.addBorders(healthySafetyIssuesWithBordersCellArray, worksheet);
+
+        worksheet.mergeCells(`B${testResults.length + 19}:C${testResults.length + 19}`);
```

Comment:
Could also use an explanation for why these cells need to be merged

Reply:
done